core/templates/hash_map.h

```diff
@@ -640,6 +642,13 @@ class HashMap {
 		capacity_index = MIN_CAPACITY_INDEX;
 	}
 
+	HashMap(std::initializer_list<KeyValue<TKey, TValue>> p_init) :
+			HashMap(MAX(p_init.size(), MIN_CAPACITY_INDEX)) {
```

Comment:
HashMap has reserve() might be good to add here also.

Reply:
These all technically called `reserve()` through the passed constructors, but I'm not sure why I didn't just do that directly in the first place. That, and the `MAX` wrapper was redundant, oops

Reply:
This is better :)

Reply:
The issue is that I failed to actually articulate is:

```cpp
#include <stdio.h>
#include <initializer_list>

struct blah {
	blah() : thing(0) {
		printf("%s\n", __PRETTY_FUNCTION__);
	}
	blah(char x) : thing(x) {
		printf("%s\n", __PRETTY_FUNCTION__);
	}

	blah(std::initializer_list<char> list): blah(5) {
		printf("%s\n", __PRETTY_FUNCTION__);
	}

	char thing;
};

int main() {
	blah b { 'a', 'b', 'c' };
}
```

```
blah::blah(char)
blah::blah(std::initializer_list<char>)
```

vs

```cpp
#include <stdio.h>
#include <initializer_list>

struct blah {
        blah() : thing(0) {
                printf("%s\n", __PRETTY_FUNCTION__);
        }
        blah(char x) : thing(x) {
                printf("%s\n", __PRETTY_FUNCTION__);
        }

        blah(std::initializer_list<char> list) {
                thing = 5;
                printf("%s\n", __PRETTY_FUNCTION__);
        }

        char thing;
};

int main() {
        blah b { 'a', 'b', 'c' };
}
```

output

```
blah::blah(std::initializer_list<char>)
```